core/src/main/scala/cats/data/Chain.scala

```diff
@@ -617,18 +620,38 @@ sealed abstract class Chain[+A] {
 
   /**
    * Remove duplicates. Duplicates are checked using `Order[_]` instance.
+   * 
+   * {{{
+   * scala> import cats.data.Chain
+   * scala> val chain = Chain(1, 2, 2, 3)
+   * scala> chain.distinct
+   * res0: cats.data.Chain[Int] = Chain(1, 2, 3)
+   * }}}
    */
-  def distinct[AA >: A](implicit O: Order[AA]): Chain[AA] = {
-    implicit val ord: Ordering[AA] = O.toOrdering
+  def distinct[AA >: A](implicit O: Order[AA]): Chain[AA] =
+    distinctBy(identity[AA])
```

Comment:
I think it may be worth it to duplicate the code to avoid pushing everything through an extra function call.

Reply:
Well, this is the code clarity VS performance dichotomy :)
Seems that Scala Library folks decided not to sweat about it too much – they simply call `distinctBy` from `distinct`.
https://github.com/scala/scala/blob/v2.13.8/src/library/scala/collection/Seq.scala#L206

But we can do the opposite, it you think it's worth it.

Reply:
Fixed.